internal/provider/provider.go

```diff
@@ -66,21 +67,26 @@ func (p *Provider) loadItems(ctx context.Context, cfg config.Config) {
 	}
 }
 
-func (p *Provider) GetSecretByType(ctx context.Context, itemName string, cfg config.Config) (string, error) {
+func (p *Provider) GetSecretByType(ctx context.Context, itemName string, cfg config.Config) (int32, string, error) {
 	item, err := p.DescribeItem(ctx, itemName, cfg)
 	if err != nil {
-		return "", err
+		return 0, "", err
 	}
+	version := item.GetLastVersion()
```

Comment:
This can be problematic if the secrets are rotated.
Let's say that our customer has a specific secret version X that they use for deployment Y
and that they create secret version X+1 for their deployment Y+1 but they do a rollback or they simply created X+1 before they wanted to deploy Y+1 then they will be forced to work with secret X+1 but with deployment Y which could cause unforeseen issues in their production. This is especially true now that the rotation is activated also on the static secret.

Reply:
Currently we don't have any options to ask specific version of secret in CSI Driver

Reply:
That's correct but we should be able to provide that information in the secret name and parse it